test(ast/estree): check span offsets are converted in ascending order in ESTree conformance tests

overlookmotel opened the pull request:
Add a Cargo feature `conformance` to `oxc_ast_visit` crate.

When this feature is enabled, it exposes a method `Utf8ToUtf16::convert_program_with_ascending_order_checks` which panics if span offsets are converted out of order.

Use this feature and method in ESTree conformance tests. This ensures:

1. `Utf8ToUtf16Converter` visitor is functioning as intended.
2. All AST types have their fields ordered in order they appear in source code.

overlookmotel commented:
* **#10887** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10887?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10887?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10887).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented on crates/oxc_ast_visit/src/utf8_to_utf16.rs:
```diff
@@ -126,20 +156,39 @@ pub struct Utf8ToUtf16Converter<'t> {
     /// # SAFETY
     /// `translations` must contain at least 2 entries.
-    unsafe fn new(translations: &'t [Translation]) -> Self {
+    #[cfg_attr(not(feature = "conformance"), expect(unused_variables))]
```
Every bloody time...

Boshen commented:
### Merge activity

* **May 13, 9:20 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
test(ast/estree): check span offsets are converted in ascending order in ESTree conformance tests (#10887)

Add a Cargo feature `conformance` to `oxc_ast_visit` crate.

When this feature is enabled, it exposes a method `Utf8ToUtf16::convert_program_with_ascending_order_checks` which panics if span offsets are converted out of order.

Use this feature and method in ESTree conformance tests. This ensures:

1. `Utf8ToUtf16Converter` visitor is functioning as intended.
2. All AST types have their fields ordered in order they appear in source code.
